src/common/charconv.cc

```diff
@@ -911,7 +911,7 @@ from_chars_result FromCharFloatImpl(const char *buffer, const int len,
   // the bias and also special-case the value 0.
   int32_t shift = (f_e2 == 0 ? 1 : f_e2) - exp_b2 - IEEE754::kFloatBias -
                   IEEE754::kFloatMantissaBits;
-  assert(shift >= 0);
+  assert(shift >= 1);
```

Comment:
Why the change of condition here?

Reply:
Tell clang-tidy that `shift` must be greater or equal to zero. I have verified the condition using `CHECK` and the `RoundTripExhaustive` JSON test.

Reply:
> Tell clang-tidy that shift must be greater or equal to zero.

But the new condition says `shift >= 1`?

Reply:
I will just put a nolint there.